registry/multi_master.go

```diff
@@ -0,0 +1,46 @@
+package registry
+
+import (
+	"fmt"
+	"os"
+)
+
+/*
+MultiMasterRegistry is an implementation of Registry in which multiple instances
+of an application are able to accept requests at the same time. Internally, this
+registry uses GroupMasterRegistry by creating a unique group ID for each
+instance of an application.
+*/
+type MultiMasterRegistry struct {
+	innerRegistry *GroupMasterRegistry
+}
+
+// NewMultiMasterRegistry creates a new MultiMasterRegistry from the provided etcd Client.
+func NewMultiMasterRegistry(key string, port int, ttl uint64) (*MultiMasterRegistry, error) {
+	group, err := makeGroupID(port)
+	if err != nil {
+		return nil, err
+	}
+
+	innerRegistry := NewGroupMasterRegistry(key, group, ttl)
```

Comment:
It's also interesting that this registry uses more complex registry to implement itself. I would factor out common logic in the `Client` for registring/deregistring and will roll out additional features in multi-master one